columnflow/selection/stats.py

```diff
@@ -0,0 +1,169 @@
+    be added. The key of each entry will result in a new field ``sum_<weight_key>``. Values should
+    consist of 2-tuples with the actual weights to be considered and a mask to be applied to it. If
+    all events should be considered, the mask can refer to an ``Ellipsis``.
+
+    Example:
+
+    .. code-block:: python
+
+        # weight map definition
+        weight_map = {
+            "mc_weight": (events.mc_weight, Ellipsis),
+            "mc_weight_selected": (events.mc_weight, results.main.event),
+        }
+
+        # usage within an exposed selector
+        # (where results are generated, and events and stats were passed by SelectEvents)
+        self[increment_stats_per_process](events, results, stats, weight_map)
+
+    Each sum of weights can also be extracted for each unique element in a so-called group, such as
+    per process id, or per jet multiplicity bin. For this purpose, a *group_map* can be defined,
+    mapping the name of a group (e.g. ``"process"`` or ``"njet"``) to a dictionary with the fields
+
+        - ``"values"``, unique values to loop over,
+        - ``"mask_fn"``, a function that is supposed to return a mask given a single value, and
+        - ``"combinations_only"`` (optional), a boolean flag (*False* by default) that decides
+            whether this group is not to be evaluated on its own, but only as part of a combination
+            with other groups (see below).
+
+    Example:
+
+    .. code-block:: python
+
+        group_map = {
+            "process"]: {
+                "values": events.process_id,
+                "mask_fn": (lambda v: events.process_id == v),
+            },
+            "njet": {
+                "values": results.x.n_jets,
+                "mask_fn": (lambda v: results.x.n_jets == v),
+            },
+        }
+
+    Based on the *weight_map* in the example above, this will result in four additional fields in
+    *stats*, i.e., ``"sum_mc_weight_per_process"``, ``"sum_mc_weight_selected_per_process"``,
+    ``"sum_mc_weight_per_njet"``, ``"sum_mc_weight_selected_per_njet"``. Each of these new fields
+    will refer to a dictionary with keys corresponding to the unique values defined in the
+    *group_map* above.
+
+    In addition, combinations of groups can be configured using *group_combinations*. It accepts a
+    sequence of tuples whose elements should be names of groups in *group_names*. As the name
+    suggests, combinations of all possible values between groups are evaluated and stored in a
+    nested dictionary.
+
+    Example:
+
+    .. code-block:: python
+
+        group_combinations = [("process", "njet")]
+
+    In this case, *stats* will obtain additional fields ``"sum_mc_weight_per_process_and_njet"``
+    and ``"sum_mc_weight_selected_per_process_and_njet"``, referring to nested dictionaries whose
+    structure depends on the exact order of group names per tuple.
+    """
+    # get the plain event mask
+    event_mask = results.main.event
```

Comment:
Oh, you're right. We should have the same mechanism here, i.e., only use `results.main.event` if no selector steps are set, and otherwise use the AND or these steps.

I will add that.

Reply:
Also, I just noticed that the event mask is not at all applied to any sum obtained through the weight map. I'll add that too.

Reply:
@mafrahm `event_mask` is now a keyword argument of `increment_stats` that only defaults to `results.main.event` now. However, when selection steps are defined, the (analysis-specific) selector can pass the mask combined from there steps now if needed.